src/vectorDBManager.ts

```diff
@@ -79,56 +109,105 @@ class VectorDBManager {
     }
   }
 
-  public static async getMemoryVectors(docHash: string): Promise<MemoryVector[] | undefined> {
+  public static async loadFile(noteFile: NoteFile, embeddingsAPI: Embeddings) {
     if (!this.db) throw new Error("DB not initialized");
+    const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 5000 })
+    const splitDocument = await textSplitter.createDocuments([noteFile.content])
+    const docVectors = await embeddingsAPI.embedDocuments(splitDocument.map((doc) => doc.pageContent))
+    const memoryVectors = docVectors.map((docVector, i) => (
+      {
+        content: splitDocument[i].pageContent,
+        metadata: {
+          ...noteFile.metadata,
+          title: noteFile.basename,
+          path: noteFile.path,
+
+        },
+        embedding: docVector,
+      }))
+    const docHash = VectorDBManager.getDocumentHash(noteFile.path);
+
+    const serializedMemoryVectors = JSON.stringify(memoryVectors);
     try {
-      const doc: VectorStoreDocument = await this.db.get(docHash);
-      if (doc && doc.memory_vectors) {
-        return JSON.parse(doc.memory_vectors);
-      }
+      // Attempt to fetch the existing document, if it exists.
+      const existingDoc = await this.db.get(docHash).catch(err => null);
+
+      // Prepare the document to be saved.
+      const docToSave = {
```

Comment:
Correct me if i'm wrong, so each `noteFile` maps to one `docToSave` with its `serializedMemoryVectors`, then could there be a case when the user switches to a different embedding model at one point, starts to index new files with that model, so they end up having a memory vector store of a bunch of `serializedMemoryVectors` embedded by different models 🤔 ? If that can happen it's quite hard to manage. I wonder what's a good way to avoid this, WDYT?

Could we add the embedding model used into some metadata here to keep track, and pop a warning when the vector store of the vault has mixed embeddings? In my mind, the vector store of the entire vault should have a unified interface that can ensure it always has the same type of embedding.

Reply:
I had not considered the case of different embedding models being used. 

I can think of two options (including the one you proposed):
- Use metadata in documents to filter by embedding model. This is simple at read time and write time. It will make it harder to use the modified time cache strategy currently implemented to limit indexing to files modified since last index
- Create a separate logical database per embedding model. We currently initialise the pouchDB with a database called `copilot_vector_stores`. We could instead use a name based on the embedding model to create multiple parallel databases for each embedding model and managed their indexing cache in the same way.

Reply:
It's a neat idea to have separate db for different embedding models, but when we have more and more embedding providers added, it creates an additional overhead to think about, and more storage. I'm inclined to a simpler setup first for now. We could add a warning message at the embedding provider dropdown, something like: "Switching embedding provider will result in re-indexing your whole vault", and then make sure we always have the same embedding model used in one index.

Reply:
I quite like that as an option if we're happy keeping it simple for now. It's nice to not over-optimise early on 😅

That being said, I just tried adding embedding providers in the metadata and it was relatively easy to refactor the code to handle using modified at time + embedding model to decide which files to embed on load.

let me know what you think 😄